frontend/components/MainPage/sections/RewardsSection/RewardsStreak.tsx

```diff
@@ -0,0 +1,54 @@
+import { RightOutlined } from '@ant-design/icons';
+import { Flex, Typography } from 'antd';
+import styled from 'styled-components';
+
+import { FireNoStreak } from '@/components/custom-icons/FireNoStreak';
+import { FireStreak } from '@/components/custom-icons/FireStreak';
+import { Pages } from '@/enums/PageState';
+import { usePageState } from '@/hooks/usePageState';
+import { useRewardsHistory } from '@/hooks/useRewardsHistory';
+
+const { Text } = Typography;
+
+const RewardsStreakFlex = styled(Flex)`
+  padding: 8px 16px;
+  background: #f2f4f9;
+  border-radius: 6px;
+  justify-content: space-between;
+  height: 40px;
+  align-items: center;
+`;
+
+const StreakIcon = ({ isOnStreak }: { isOnStreak: boolean }) => {
+  if (isOnStreak) return <FireStreak />;
+  return <FireNoStreak />;
+};
+const StreakText = ({ streak }: { streak: number }) => {
+  if (streak > 0) return <div>{streak} day streak</div>;
+  return <div>No streak</div>;
+};
+
+export const RewardsStreak = () => {
+  const { goto } = usePageState();
+  const { latestRewardStreak, isLoading } = useRewardsHistory();
+
+  if (isLoading || latestRewardStreak === null) return null;
+
+  return (
+    <RewardsStreakFlex>
+      <span style={{ display: 'inline-flex', gap: 8 }}>
```

Comment:
it feels a bit like overengineering to have two components StreakIcon and StreakText, where it can be only one StreakSomething that includes this span as well, wdyt?